src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -151,15 +151,21 @@ public Builder addNamed(String format, Map<String, ?> arguments) {
           formatParts.add(format.substring(p, nextP));
           p = nextP;
         }
-        Matcher matcher = NAMED_ARGUMENT.matcher(format.subSequence(p, format.length()));
-        if (matcher.matches()) {
+
+        Matcher matcher = null;
+        int colon = format.indexOf(':', p);
+        if (colon != -1) {
+          int endIndex = Math.min(colon + 2, format.length());
+          matcher = NAMED_ARGUMENT.matcher(format.substring(p, endIndex));
```

Comment:
I also realize that `subSequence` was not implemented how I had originally hoped (as an iterator) and instead it's just a straight `substring` call... So, we do want to retain the colon lookup to avoid creating a substring with the remainder of the code block each time.